### PR TITLE
Skip broken shares when transferring ownership

### LINKED ISSUE #26524: Transfer ownership don't bail out on error
See https://github.com/owncloud/core/issues/26523.

In case a share couldn't be transfered, log it but continue working on the next shares.
Display a list of failed shares.

### LINKED ISSUE #26525: Transfer ownership don't fail on shares pointing to trashed files
### Steps

1. Create three users "user1", "user2", "user3"
1. Login as "user1"
1. Create a folder "delete/this"
1. Share "this" with "user3"
1. Delete folder "delete"
1. Transfer ownership of "user1" to "user2"

### Expected result
Files transferred, trashbin and trashbin shares ignored (or deleted?).

### Actual result
```
± % sudo -uwwwrun ./occ files:transfer-ownership user1 user2
vincent's password:
Analysing files of user1 ...
    0 [>---------------------------]
Collecting all share information for files and folder of user1 ...
    1 [============================]
Transferring files to user2/files/transferred from user1 on 2016-11-02T14:06:42+00:00 ...
Restoring shares ...

                                 
  [OCP\Files\NotFoundException]  
```

This is also likely to happen for any orphaned share pointing to invalid files.
In these cases, best is to catch the "NotFoundException" and ignore.

### Versions
Observed on OC 9.0.5

@DeepDiver1975 @owncloud/qa

### PR DESCRIPTION
**Pull request by PVince81**
## Related Issue
Fixes https://github.com/owncloud/core/issues/26524
Fixes https://github.com/owncloud/core/issues/26525

## Motivation and Context
Because aborting isn't useful as it would prevent legitimate subsequent shares to be migrated.

## How Has This Been Tested?
See steps from the tickets

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### Backports
- [ ] stable9.1
- [ ] stable9

@DeepDiver1975

### CODE REVIEW

**Comment by PVince81** [comment]
Integration tests will be added as part of https://github.com/owncloud/core/pull/26543

**Comment by PVince81** [comment]
@DeepDiver1975 @jvillafanez please review

File: apps/files/lib/Command/TransferOwnership.php [hunk 1]
```diff
@@ -220,22 +220,28 @@ private function restoreShares(OutputInterface $output) {
...
-					$share->setSharedBy($this->destinationUser);
-				}
+			try {
+				if ($share->getSharedWith() === $this->destinationUser) {
+					// Unmount the shares before deleting, so we don't try to get the storage later on.
+					$shareMountPoint = $this->mountManager->find('/' . $this->destinationUser . '/files' . $share->getTarget());
+					if ($shareMountPoint) {
+						$this->mountManager->removeMount($shareMountPoint->getMountPoint());
+					}
+					$this->shareManager->deleteShare($share);
```
Comment (DeepDiver1975) [inline apps/files/lib/Command/TransferOwnership.php]: deleteShare throws ShareNotFound according to PHPDoc .... but I don't see this being throws in the implementation :puke:

**Comment by DeepDiver1975** [comment]
:+1:

File: apps/files/lib/Command/TransferOwnership.php (hunk 1 above)
Comment (PVince81) [inline apps/files/lib/Command/TransferOwnership.php]: :sad:

**Comment by PVince81** [comment]
stable9: https://github.com/owncloud/core/pull/26575
stable9.1: https://github.com/owncloud/core/issues/26575

**Pull request merged**